src/cdk-experimental/ui-patterns/behaviors/list-typeahead/list-typeahead.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import {signal, Signal} from '@angular/core';
+import {ListNavigationItem, ListNavigation} from '../list-navigation/list-navigation';
+
+/** The required properties for typeahead items. */
+export interface ListTypeaheadItem extends ListNavigationItem {
+  /** The text used by the typeahead search. */
+  searchTerm: Signal<string>;
+}
+
+/** The required inputs for list typeahead. */
+export interface ListTypeaheadInputs {
+  /** The amount of time before the typeahead search is reset. */
+  typeaheadDelay: Signal<number>;
+}
+
+/** Controls typeahead for a list of items. */
+export class ListTypeahead<T extends ListTypeaheadItem> {
+  /** A reference to the timeout for resetting the typeahead search. */
+  timeout?: any;
```

Comment:
```suggestion
  timer: ReturnType<typeof setTimeout> | undefined;
```
by defining this in terms of the return type of `setTimeout`, it's correct in all environments and still type safe